internal/vm/importers/composite.go

```diff
@@ -0,0 +1,28 @@
+package importers
+
+import (
+	"fmt"
+
+	"github.com/google/go-jsonnet"
+)
+
+// CompositeImporter tries multiple extended importers in sequence for a given path
```

Comment:
You need to be really careful about canonical (foundAt) paths returned by the component importers. If all importers return compatible paths then it's fine, but this API allows combining totally unrelated importers and there are a couple ways it could go wrong.

1) Violating the requirement to always return the same contents for the same foundAt (imporer A returns content "a" and foundAt "path" for path "a", importer B returns content "b" and foundAt "path" for path "b").  https://github.com/google/go-jsonnet/blob/v0.16.0/imports.go#L39
2) Path returned by one importer may confuse another importer when passed as importedFrom.

I think it's worth documenting,

It looks like your globimporter handles it by creating fake filesystem paths. This should be good enough.